src/class-savvii-dashboard.php

```diff
@@ -162,13 +155,82 @@ function warpdrive_dashboard() {
+                                        </div>
+                                    </div>
+                                    <div class='column'>
+                                        <h3>Error log preview</h3>
+                                        <div class='code'>
+                                            <?php $logreader->print_lines( 'error', 10 ); ?>
+                                        </div>
+                                    </div>
+                                </div>
+                            <?php
+                            } else {
+                                echo "<p>Only users with the Super Admin role are allowed to read server logs</p>";
+                            }
+                            ?>
                         </div>
                     </div>
                     <!-- /Read server logs -->
+                    <!-- Statistics -->
+                    <div class="postbox">
+                        <h2 class="hndle">Statistics</h2>
+                        <div class="inside">
+                            <div class="row">
+                                <?php
+                                $errors = $logreader->find_match_in_log( 'access', '/HTTP\/(?:1\.1|2)\"\s+(50[24])/i');
+                                $errors_502 = 0;
+                                $errors_504 = 0;
+                                foreach ($errors as $error) {
+                                    if (strpos($error, '502')) {
+                                        $errors_502++;
+                                    } elseif (strpos($error, '504')) {
+                                        $errors_504++;
+                                    }
+                                }
+
+                                //$errors_502 = $logreader->find_match_in_log( 'access', '/HTTP\/(?:1\.1|2)\"\s+502/i');
+                                //$errors_504 = $logreader->find_match_in_log( 'access', '/HTTP\/(?:1\.1|2)\"\s+504/i');
+                                ?>
+                                <div class="column">
+                                    <p>WarpDrive detected <?php echo ' ' . $errors_502 . ' ';?> recent 502 (Bad Gateway) error(s)</p>
+                                    <p>WarpDrive detected <?php echo ' ' . $errors_504 . ' ';?> recent 504 (Gateway Timeout) error(s)</p>
+                                </div>
+                                <?php
+                                $results = Database::get_wp_table_sizes(10);
+                                ?>
+                                <div class="column">
+                                    <table>
+                                        <tr>
+                                            <th>Table</th>
+                                            <th>Size (MB)</th>
+                                        </tr>
+                                    <?php foreach($results as $row) { ?>
```

Comment:
As `$results` is quite a vague term for the info you're retrieving and you're only using it once, you can use the function call directly in the `for`.
```suggestion
                                    <?php foreach(Database::get_wp_table_sizes(10) as $row) { ?>
```